src/caffeine/program_termination_s.f90

```diff
@@ -125,31 +89,8 @@ subroutine inner_caf_error_stop_integer(stop_code) bind(C, name = "inner_caf_err
     end if
 
     call caf_decaffeinate(exit_code) ! does not return
-
   end subroutine
 
-  pure function f_c_string(f_string) result(c_string)
-    character(len=*), intent(in) :: f_string
-    character(len=1,kind=c_char) :: c_string(len(f_string))
-
-    integer :: i
-
-    do concurrent (i = 1:len(f_string))
-      c_string(i) = f_string(i:i)
-    end do
-  end function
-
-  pure function c_f_string(c_string, length) result(f_string)
-    integer(c_int), intent(in) :: length
-    character(len=1,kind=c_char), intent(in) :: c_string(length)
-    character(len=length) :: f_string
-
-    integer :: i
-    do concurrent (i = 1:length)
-      f_string(i:i) = c_string(i)
-    end do
-  end function
```

Comment:
We may end up needing these string-conversion utilities later (e.g. to fix the implementation of co_min/co_max on strings), but we can always restore them when needed (and they really belong in prif_private_s.f90 instead of here).